go/posix-router/main.go

```diff
@@ -40,6 +45,60 @@ import (
 
 var globalCfg config.Config
 
+type rateLimitingServer struct {
+	colgrpc.UnimplementedRateLimiterServiceServer
+	dp *router.DataPlane
+}
+
+func (r *rateLimitingServer) AddRateLimit(ctx context.Context, req *colpb.AddRateLimitRequest) (*colpb.Success, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	r.dp.SyncRateLimiter.Ratelimiter.AddRatelimit(req.Identifier, req.Rate, req.Cbs, time.Now())
+	return &colpb.Success{}, nil
+}
+
+func (r *rateLimitingServer) GetBurstSize(ctx context.Context, req *colpb.GetterForIdentifierRequest) (*colpb.GetBurstSizeResponse, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	cbs, err := r.dp.SyncRateLimiter.Ratelimiter.GetBurstSize(req.Identifier)
+	return &colpb.GetBurstSizeResponse{Cbs: cbs}, err
+}
+
+func (r *rateLimitingServer) GetBurstSizeAndRate(ctx context.Context, req *colpb.GetterForIdentifierRequest) (*colpb.GetBurstSizeAndRateResponse, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	cbs, rate, err := r.dp.SyncRateLimiter.Ratelimiter.GetBurstSizeAndRate(req.Identifier)
+	return &colpb.GetBurstSizeAndRateResponse{Cbs: cbs, Rate: rate}, err
+}
+
+func (r *rateLimitingServer) GetRate(ctx context.Context, req *colpb.GetterForIdentifierRequest) (*colpb.GetRateResponse, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	rate, err := r.dp.SyncRateLimiter.Ratelimiter.GetRate(req.Identifier)
+	return &colpb.GetRateResponse{Rate: rate}, err
+}
+
+func (r *rateLimitingServer) SetBurstSize(ctx context.Context, req *colpb.SetBurstSizeRequest) (*colpb.Success, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	err := r.dp.SyncRateLimiter.Ratelimiter.SetBurstSize(req.Identifier, req.Cbs)
+	return &colpb.Success{}, err
+}
+
+func (r *rateLimitingServer) SetBurstSizeAndRate(ctx context.Context, req *colpb.SetBurstSizeAndRateRequest) (*colpb.Success, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	err := r.dp.SyncRateLimiter.Ratelimiter.SetBurstSizeAndRate(req.Identifier, req.Cbs, req.Rate)
+	return &colpb.Success{}, err
+}
+
+func (r *rateLimitingServer) SetRate(ctx context.Context, req *colpb.SetRateRequest) (*colpb.Success, error) {
+	r.dp.SyncRateLimiter.Lock()
+	defer r.dp.SyncRateLimiter.Unlock()
+	err := r.dp.SyncRateLimiter.Ratelimiter.SetRate(req.Identifier, req.Rate)
+	return &colpb.Success{}, err
+}
+
```

Comment:
There is no need to offer that many gRPC calls. I think a set and a get would be sufficient.